core/src/main/java/org/apache/iceberg/io/ResolvingFileIO.java

```diff
@@ -122,6 +124,15 @@ public void initialize(Map<String, String> newProperties) {
     close(); // close and discard any existing FileIO instances
```

Comment:
This method can be called multiple times. You should probably clear `schemeToFileIo`, either here or better yet, inside the `close()` method.